File: LotachampCore/Lotachamp.Api/ClientApp/src/app/models/score.ts
import { Participant } from "./participant.model";
import { SportEvent } from "./sport-event.model";
import { Tour } from "./tour.model";

export class Score {
  constructor(
    public scoreId?: number,
    public tour?: Tour,
    public participant?: Participant,
    public sportEvent?: SportEvent,
    public scoreDate?: Date,
    public resultValue?: number,
    public resultUnit?: string,
    public notes?: string,
    public points?: number,
    public rank?: number,
    public created?: Date,
    public createdBy?: string,
  ) { }
}

File: LotachampCore/Lotachamp.Model/Entities/Score.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lotachamp.Model.Entities
{
    public class Score : BaseEntity
    {
        public Score()
        {
            Notes = string.Empty;
            Created = DateTime.Now;
            Pictures = new HashSet<Picture>();
        }

        public Guid ScoreId { get; set; }
        public int SportId { get; set; }
        public Guid ParticipantId { get; set; }
        public int ResultValue { get; set; }
        public DateTime ScoreDate { get; set; }
        public string Notes { get; set; }

        public virtual ICollection<Picture> Pictures { get; private set; }
        public virtual Sport Sport { get; set; }
        public virtual Participant Participant { get; set; }
    }
}
